app/javascript/article-form/actions.js

```diff
@@ -13,7 +13,13 @@ export function previewArticle(payload, successCb, failureCb) {
     }),
     credentials: 'same-origin',
   })
-    .then((response) => response.json())
+    .then((response) => {
```

Comment:
I implemented your suggestion @fdoxyz, so the failure callback kicks in.

Reply:
Awesome! The failure callback is kicking in but the error isn't showing because I think it needs to be parsed instead of returning a `response` object.

This is what I see when I pass in a bad tag (with an error)

https://share.getcloudapp.com/WnuBEg8X

Reply:
I was following the existing pattern used for errors, but if the processed_html is empty, it's not really an error is it? I just check for the HTTP status and do like in `handleArticleError` where they set the errors to the response. I agree a response as an error is a bit bizarre though.

https://github.com/forem/forem/blob/2a5fe650998e989faf2caf3a65d4b39911575eea/app/javascript/article-form/articleForm.jsx#L247-L256

Reply:
Just saw your video. Took a few minutes to upload it I guess. I see what you're seeing now. Ley me just see how that was working prior to the fix.

Reply:
OK, I see what I need to do. Small change coming up @fdoxyz.

Reply:
Throwing the `response.json()` fixes the issue. This is what it was doing previously but on the success callback. The only issue I can see with this is if there is no JSON returned because of an error like a 500, but that would have blown up the way it did before I believe. I'd have to test that.

Outside the scope of this bug fix, but maybe revisit the error handling in this.

Reply:
Thanks for chiming in here @nickytonline! The async/await + Promise worlds don't mix well in my head 😆

> The only issue I can see with this is if there is no JSON returned because of an error like a 500, but that would have blown up the way it did before I believe

> Outside the scope of this bug fix, but maybe revisit the error handling in this.

I agree and also think it would be handled by the `.catch(failureCb)` in `article-form/actions.js` fetch.

As for revisiting the error handling, I think that's a great idea and you may have mentioned somewhere about covering these cases with Cypress E2E tests, that follow-up would be awesome for sure 🙌🏼